timesketch/lib/analyzers/expert_sessionizers_test.py

```diff
@@ -2,102 +2,47 @@
 
+import unittest
 import mock
 
 from timesketch.lib.analyzers.expert_sessionizers import \
     WebActivitySessionizerSketchPlugin
-from timesketch.lib.analyzers.sessionizer_test import _create_mock_event
+from timesketch.lib.analyzers.base_sessionizer_test import BaseSessionizerTest
+from timesketch.lib.analyzers.base_sessionizer_test import _create_mock_event
 from timesketch.lib.testlib import BaseTest
 from timesketch.lib.testlib import MockDataStore
 
-
-class TestWebActivitySessionizerPlugin(BaseTest):
+class TestWebActivitySessionizerPlugin(BaseTest, BaseSessionizerTest):
     """Tests the functionality of the web activity sessionizing sketch
     analyzer."""
-    @mock.patch('timesketch.lib.analyzers.interface.ElasticsearchDataStore',
-                MockDataStore)
-    def test_analyzer(self):
-        """Test basic analyzer functionality."""
-        index = 'test_index'
-        sketch_id = 1
-        analyser = WebActivitySessionizerSketchPlugin(index, sketch_id)
-        self.assertIsInstance(analyser, WebActivitySessionizerSketchPlugin)
-        self.assertEqual(index, analyser.index_name)
-        self.assertEqual(sketch_id, analyser.sketch.id)
+    analyzer_class = WebActivitySessionizerSketchPlugin
 
     @mock.patch('timesketch.lib.analyzers.interface.ElasticsearchDataStore',
                 MockDataStore)
-    def test_same_session(self):
-        """Test multiple events in the same session are allocated and labelled
-        correctly."""
-        with mock.patch.object(WebActivitySessionizerSketchPlugin,
-                               'event_stream',
-                               return_value=_create_mock_event(
-                                   0, 2, time_diffs=[200])):
+    def test_event_type(self):
+        """Test the mocking of events returns an event stream that matches the
+        query for the analyzer."""
+        with mock.patch.object(
+            self.analyzer_class,
+            'event_stream',
+            return_value=_create_mock_event(
+                0,
+                2,
+                source_attrs={'source_short': 'WEBHIST'})):
             index = 'test_index'
             sketch_id = 1
-            analyser = WebActivitySessionizerSketchPlugin(index, sketch_id)
-            message = analyser.run()
+            analyzer = self.analyzer_class(index, sketch_id)
+            message = analyzer.run()
             self.assertEqual(
                 message,
                 'Sessionizing completed, number of session created: 1')
 
             ds = MockDataStore('test', 0)
             event1 = (ds.get_event('test_index', '0', stored_events=True))
-            self.assertEqual(event1['_source']['session_id'],
-                             {'web_activity': 1})
-            # checking event with id '101' as 100 events have been inserted
-            # as 'padding' (see _create_mock_event())
+            self.assertEqual(event1['_source']['source_short'], 'WEBHIST')
```

Comment:
I think it's worth keeping the check that the session_id also matches what you expect *in addition* to checking for the source. Sorry if I wasn't clear :)